### PR TITLE
Clean up the abstract file handler

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: src/handler.ts [hunk 1]
```diff
@@ -116,39 +112,55 @@ abstract class AbstractFileHandler implements IMessageFilter {
...
+      return Promise.resolve(void 0);
     }
     let model = this._getModel(widget);
-    return this.getState(widget, model).then(contents => {
-      return this.manager.save(model.path, contents).then(contents => {
-        widget.title.className = widget.title.className.replace(DIRTY_CLASS, '');
-        return contents;
-      });
+    return this.getSaveOptions(widget, model).then(opts => {
+      return this.manager.save(model.path, opts)
```
Comment (afshin) [inline src/handler.ts]: Needs a `;` at the end of the line.

**Comment by blink1073** [comment]
Wait a sec on this.

**Comment by blink1073** [comment]
Ready.

**Comment by dwillmer** [comment]
:+1:

**Pull request merged**